src/fairseq2/recipes/eval/__init__.py

```diff
@@ -0,0 +1,28 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from fairseq2.recipes.cli import Cli, CliGroup, RecipeCommandHandler
+from fairseq2.recipes.eval.asr import load_wav2vec2_asr_evaluator
+from fairseq2.recipes.eval.configs import hf_presets
+
+
+def _add_wav2vev2_asr_eval_cli(group: CliGroup) -> None:
+    handler = RecipeCommandHandler(
+        load_wav2vec2_asr_evaluator,
+        preset_configs=hf_presets,
+        default_preset="librispeech_asr",
+    )
+    group.add_command(
+        "wav2vec2-asr",
+        handler,
+        help="evaluate a wav2vec 2.0 ASR model in downstream benchmark",
+    )
+
+
+def _setup_eval_cli(cli: Cli) -> None:
+    group = cli.add_group("eval", help="Evaluate fairseq2 models in downstream tasks")
+
+    _add_wav2vev2_asr_eval_cli(group)
```

Comment:
See my other comments. It might be better that we add sub-command conditionally on the availability of huggingface's datasets or evaluate libs

Reply:
I’m still getting familiar with handling import exceptions in this way, so I appreciate your patience.

Do you expect something like this?

```python
try:
    _add_wav2vev2_asr_eval_cli(group)
except ImportError as e:
    # Log the reason for skipping this command
```

However, wouldn't this approach raise an exception for all sub-commands since the main idea behind this eval recipe is to use huggingface evaluate

Reply:
> In other words, no sub-commands will be available if hf libs are not installed unless this recipe is going to be extended with commands that doesn't require hf libs

No, fairseq2.recipes.eval is an umbrella to test fairseq2 models , with or without using huggingface datasets. The public version here relies on public benchmarks that are available via huggingface, but we can extend to other benchmarks / metrics too.

And only the hf-based recipes are isolated within the try-except block